spec/omniauth/strategies/saml_spec.rb

```diff
@@ -207,6 +207,29 @@ def post_xml(xml=:example_response, opts = {})
       end
     end
 
+    context "when using custom user id attribute" do
+      before :each do
+        saml_options[:idp_cert_fingerprint] = "3B:82:F1:F5:54:FC:A8:FF:12:B8:4B:B8:16:61:1D:E4:8E:9B:E2:3C"
+        saml_options[:uid_attribute] = "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress"
+        post_xml :custom_attributes
+      end
+
+      it "should return user id attribute" do
+        expect(auth_hash[:uid]).to eq("user@example.com")
+      end
+    end
+
+    context "when using custom user id attribute, but it is missing" do
+      before :each do
+        saml_options[:uid_attribute] = "missing_attribute"
+        post_xml
+      end
+
+      it "should not return an authenticated session" do
+        expect(auth_hash).to eq(nil)
```

Comment:
That was what I had assumed, but upon a closer look, I realized that it was actually handled by [this exception handler in `strategies/saml.rb`](https://github.com/omniauth/omniauth-saml/blob/v1.7.0/lib/omniauth/strategies/saml.rb#L79).

With that in mind I see how to add a more comprehensive test for this error, so I have now implemented that.